src/application/gui/HighscoreLeaderboard.java

```diff
@@ -0,0 +1,108 @@
+package application.gui;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.control.ListView;
+
+public class HighscoreLeaderboard extends ListView<HighscoreLeaderboard.Entry>
+{
+    private ObservableList<Entry> entries;
+    private Map<Integer, Entry> addedEntries;
```

Comment:
im pretty confident you could replace most of those collections with a single TreeSet